pageserver/src/tenant/timeline.rs

```diff
@@ -4663,8 +4718,11 @@ impl Timeline {
         let mut guard = self.layers.write().await;
 
         // FIXME: we could add the images to be uploaded *before* returning from here, but right
-        // now they are being scheduled outside of write lock
-        guard.track_new_image_layers(&image_layers, &self.metrics);
+        // now they are being scheduled outside of write lock; current way is inconsistent with
+        // compaction lock order.
+        guard
+            .open_mut()?
+            .track_new_image_layers(&image_layers, &self.metrics);
```

Comment:
How confident are you this won't cause deadlocks?

Reply:
it has been like this for a long time, I am doing somw drive by comment adjustment here.